src/Servers/Kestrel/Transport.Sockets/src/Internal/SocketConnection.cs

```diff
@@ -48,6 +48,7 @@ internal sealed class SocketConnection : TransportConnection
             MemoryPool = memoryPool;
             _trace = trace;
             _waitForData = waitForData;
+            Socket = _socket;
```

Comment:
This is the only place we need the feature to exist.

Reply:
Thanks @davidfowl for comments, was waiting for this only. I will get this changes reflected in my next commit! 😇

Thanks again!

Reply:
Hi @davidfowl, I addressed your comments in my latest commit, but now the problem is, I cannot able to access `Socket` inside `SocketConnection.cs` constructor, as `TransportConnection` does not implement `IConnectionSocketFeature`. TransportConnection is built by 3 partial classes! 
![image](https://user-images.githubusercontent.com/17148381/110954057-13659e00-836e-11eb-9153-f52cbbdda1d9.png)


And also based on the issue that I linked in my PR description, we have expose the underlying socket to users via `IConnectionSocketFeature`, so somehow somewhere we need to add that feature to `FeatureCollection` as well correct? 

Now the question is how can I assign `Socket` variable which is insider my `IConnectionSocketFeature` in the constructor of `SocketConnection.cs`.